core/services/workflows/delegate_test.go

```diff
@@ -6,57 +6,150 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/smartcontractkit/chainlink/v2/core/services/workflows"
+	"github.com/smartcontractkit/chainlink/v2/core/testdata/testspecs"
 )
 
 func TestDelegate_JobSpecValidator(t *testing.T) {
 	t.Parallel()
 
 	var tt = []struct {
-		name  string
-		toml  string
-		valid bool
+		name           string
+		workflowTomlFn func() string
+		valid          bool
 	}{
 		{
-			"valid spec",
-			`
-type = "workflow"
-schemaVersion = 1
-workflowId = "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
-workflowOwner = "00000000000000000000000000000000000000aa"
-workflowName = "test"
-`,
+			"missing workflow field",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				return testspecs.GenerateWorkflowSpec(id, owner, "test", "").Toml()
+			},
+			false,
+		},
+
+		{
+			"null workflow",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				return testspecs.GenerateWorkflowSpec(id, owner, "test", "{}").Toml()
+			},
+			false,
+		},
+
+		{
+			"missing name",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				wf := `
+triggers: []
+consensus: []
+targets: []
+`
+				return testspecs.GenerateWorkflowSpec(id, owner, "", wf).Toml()
+			},
+			false,
+		},
+
+		{
+			"minimal passing workflow",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				wf := `
+triggers: []
+consensus: []
+targets: []
+`
+				return testspecs.GenerateWorkflowSpec(id, owner, "test", wf).Toml()
+			},
 			true,
 		},
+
+		// Taken from jobs controller test, as we want to fail early without a db / slow test dependency
+		{
+			"valid full spec",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				workflow := `
+triggers:
+  - id: "mercury-trigger@1.0.0"
+    config:
+      feedIds:
+        - "0x1111111111111111111100000000000000000000000000000000000000000000"
+        - "0x2222222222222222222200000000000000000000000000000000000000000000"
+        - "0x3333333333333333333300000000000000000000000000000000000000000000"
+
+consensus:
```

Comment:
Should this have and validate `report_id` field?

Reply:
`report_id` sounds like its specific to the `offchain_reporting` capability `input` or `config`. Capabiltiies are responsible for validating their own `inputs` and `config`. The workflow spec validator treats them as maps of `any ` type.

Reply:
Yep, we've put that inside the OCR capability's config field